src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5085,7 +5085,9 @@ and TcPatLongIdentActivePatternCase warnOnUpper (cenv: cenv) (env: TcEnv) vFlags
 
             if dtys.Length = args.Length + 1 &&
                 ((isOptionTy g retTy && isUnitTy g (destOptionTy g retTy)) ||
-                (isValueOptionTy g retTy && isUnitTy g (destValueOptionTy g retTy))) then
+                (isValueOptionTy g retTy && isUnitTy g (destValueOptionTy g retTy))) ||
+                // `bool` partial AP always be treated as `unit option`
```

Comment:
It is possible to treat `unit option`/`unit voption`-returning partial active patterns as either nullary or unary (or $(n-1)$&#x200b;-ary, for parameterized partial active patterns): a match on a pattern `val (|P|_|) : 'a -> unit option` can be written `P`, `P ()`, `P _`, or `P a`; likewise `val (|R|_|) : 'a -> 'b -> unit option` can be written `R a`, `R a ()`, `R a _`, or `R a b`.

Does this mean that it will be possible to do the same with `bool`-returning partial active patterns?

Reply:
For `val (|P|_|) : _ -> bool`, only allow `P`

For `val (|P|_|) : _ -> _ -> bool`, only allow `P a`

That said the argument count of `P` should always match the parameter count.